src/main/java/uk/gov/hmcts/reform/userprofileapi/service/impl/IdamServiceImpl.java

```diff
@@ -69,32 +81,47 @@ public IdamRolesInfo fetchUserById(String id) {
     public IdamRolesInfo updateUserRoles(List roleRequest, String userId) {
 
         ResponseEntity<Object> responseEntity;
-        Response response;
+        Response response = null;
         try {
             response = idamClient.updateUserRoles(roleRequest, userId);
             responseEntity = JsonFeignResponseHelper.toResponseEntity(response, getResponseMapperClass(response,
                     null));
         } catch (FeignException ex) {
             responseEntity = ResponseEntity.status(gethttpStatusFromFeignException(ex)).build();
         }
-
-        return new IdamRolesInfo(responseEntity);
+        IdamRolesInfo result = getIdamRolesInfo(responseEntity, response);
+        return result;
     }
 
     @Override
     public IdamRolesInfo addUserRoles(Set roleRequest, String userId) {
 
         ResponseEntity<Object> responseEntity;
-        Response response;
+        Response response = null;
         try {
             response = idamClient.addUserRoles(roleRequest, userId);
             responseEntity = JsonFeignResponseHelper.toResponseEntity(response, getResponseMapperClass(response,
                     null));
         } catch (FeignException ex) {
             responseEntity = ResponseEntity.status(gethttpStatusFromFeignException(ex)).build();
         }
+        IdamRolesInfo result = getIdamRolesInfo(responseEntity, response);
+        return result;
+    }
+
+    IdamRolesInfo getIdamRolesInfo(ResponseEntity<Object> responseEntity, Response response) {
+        IdamRolesInfo result = new IdamRolesInfo(responseEntity);
+
+        Optional<Response> respOptional = Optional.ofNullable(response);
+        if (respOptional.isPresent()) {
+            HttpStatus httpStatus = HttpStatus.valueOf(response.status());
+            if (httpStatus.is5xxServerError()) {
+                result.setResponseStatusCode(HttpStatus.UNAUTHORIZED);
+                result.setStatusMessage(IdamStatusResolver.MISSING_TOKEN);
```

Comment:
All 5xx cannot be due to missing token. This message could mislead the users. Can you please send appropriate message?